### PR TITLE
Add ssl_password_file directive to support encrypted ssl keys

### PR DESCRIPTION
**Pull request by joernott**
#### Pull Request (PR) description
This pull request adds support for the ssl_password_file directive of nginx. This directive is needed, if the ssl key file is password encrypted (should be standard nowadays).

#### This Pull Request (PR) fixes the following issues
n/a (I didn't create an issue first but tried to not only raise a request but also deliver the solution)

### CODE REVIEW

**Comment by joernott** [comment]
Note: I designed this change to be non-breaking

File: manifests/resource/server.pp [hunk 1]
```diff
@@ -187,6 +188,7 @@
   Optional[String] $ssl_session_ticket_key                                       = undef,
   Optional[String] $ssl_trusted_cert                                             = undef,
   Optional[Integer] $ssl_verify_depth                                            = undef,
+  Optional[String] $ssl_password_file                                            = undef,
```
Comment (bastelfreak) [inline manifests/resource/server.pp]: Can you please update the type?
```suggestion
  Optional[Stdlib::Absolutepath] $ssl_password_file                              = undef,
```

**Comment by bastelfreak** [comment]
Hi @joernott, thanks for the PR. Can you please add tests to it?

**Comment by joernott** [comment]
I did not find where you get your certificates from. Or is a simple test like "I added the directive and it shows up in the config file" sufficient?

**Comment by bastelfreak** [comment]
It's okay to add an rspec test that verifies the content of the file.

**Comment by bastelfreak** [comment]
Thanks for the awesome acceptance test!

**Pull request merged**